### PR TITLE
chore(RHTAPWATCH-895): add BYO Quay docs

### PR DESCRIPTION
**Pull request by ernesgonzalez33**

### CODE REVIEW

**Comment by ernesgonzalez33** [comment]
@arewm can I get your review here?

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 1]
```diff
@@ -119,6 +119,26 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
...
+== Bring your own Quay repository to the build pipeline
+
+By default, all pipelines push the images to the `quay.io/redhat-user-workloads` repository. It is not mandatory to use this Quay repo, so if you want to use your own Quay repo to control user permissions, you can do this by updating the `output-image` parameter in the pipelines.
+
+.Procedure
+
+. Go to the GitHub repo of your component.
+. In each .yaml file in the .tekton directory, under params, locate the param named output-image:
+.. Change it to the new repository. For example: `quay.io/my-new-quay-repo/my-app:on-pr-{\{revision}}` for the `pull-request` pipeline, and `quay.io/my-new-quay-repo/my-app:{\{revision}}` for the `push` pipeline. This is a best practice to know which images come from a Pull Request and which don't.
+. You will need a secret to push the images to your Quay repo.
```
[the remark this answers is not in the document]
Comment (ernesgonzalez33) [reply docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:128]: That makes a whole lot of sense! Updated

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 2]
```diff
@@ -119,6 +119,26 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
...
+.. Change it to the new repository. For example: `quay.io/my-new-quay-repo/my-app:on-pr-{\{revision}}` for the `pull-request` pipeline, and `quay.io/my-new-quay-repo/my-app:{\{revision}}` for the `push` pipeline. This is a best practice to know which images come from a Pull Request and which don't.
+. You will need a secret to push the images to your Quay repo. 
+.. Follow Quay's link:https://docs.quay.io/glossary/robot-accounts.html[docs] to create a robot account for your repo.
+.. Give push access to your repository to the robot account.
+.. Retrieve the `.dockercfg` from the "Docker Configuration" menu on your robot account credentials page.
+.. Follow the steps to link:proc_creating-secrets-for-your-builds.adoc[create a secret for your build] and do it as an "Image Pull Secret".
+
+[NOTE] 
+====
+You might need to delete the `"auths": {` and one `}` at the end of the docker config file for the secret to work. Until this https://issues.redhat.com/browse/KFLUXBUGS-1160[bug] is resolved.
```
[the remark this answers is not in the document]
Comment (ernesgonzalez33) [reply docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc]: Added the example and removed the "might" word. About updating the bug: how do you think is the best way? A comment? A sub-task? Changing the description?

[the remark this answers is not in the document]
Comment (arewm) [reply docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc]: Changing the description to make it part of the acceptance criteria sounds reasonable.

**Review by arewm — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (ernesgonzalez33) [reply docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc]: @arewm added!

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 3]
```diff
@@ -119,6 +119,52 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
 .. Under `.taskRef.params`, set `name` to `buildah-6gb`.
 .. Under `.taskRef.params`, set `bundle` to `quay.io/redhat-appstudio-tekton-catalog/task-buildah-6gb:0.1`.
 
+== Bring your own Quay repository to the build pipeline
+
+By default, all pipelines push the images to the `quay.io/redhat-user-workloads` repository. It is not mandatory to use this Quay repo, so if you want to use your own Quay repo to control user permissions, you can do this by updating the `output-image` parameter in the pipelines.
+
+.Procedure
+
+. You will need a secret to push the images to your Quay repo.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:128]:
```suggestion
. You need a secret to push the images to your Quay repo. 
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 4]
```diff
@@ -119,6 +119,52 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
...
 .. Under `.taskRef.params`, set `bundle` to `quay.io/redhat-appstudio-tekton-catalog/task-buildah-6gb:0.1`.
 
+== Bring your own Quay repository to the build pipeline
+
+By default, all pipelines push the images to the `quay.io/redhat-user-workloads` repository. It is not mandatory to use this Quay repo, so if you want to use your own Quay repo to control user permissions, you can do this by updating the `output-image` parameter in the pipelines.
+
+.Procedure
+
+. You will need a secret to push the images to your Quay repo. 
+.. Follow Quay's link:https://docs.quay.io/glossary/robot-accounts.html[docs] to create a robot account for your repo.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:129]:
```suggestion
.. Follow Quay's link:https://docs.quay.io/glossary/robot-accounts.html[documentation] to create a robot account for your repository.
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 5]
```diff
@@ -119,6 +119,52 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
...
+.. Give push access to your repository to the robot account.
+.. Retrieve the `.dockercfg` from the "Docker Configuration" menu on your robot account credentials page.
+.. Follow the steps to link:proc_creating-secrets-for-your-builds.adoc[create a secret for your build] and do it as an "Image Pull Secret".
+
+[NOTE] 
+====
+You need to delete the `"auths": {` and one `}` at the end of the docker config file for the secret to work. Until this https://issues.redhat.com/browse/KFLUXBUGS-1160[bug] is resolved.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:136]:
```suggestion
You need to delete the `"auths": {` and one `}` at the end of the docker config file for the secret to work, pending the resolution of https://issues.redhat.com/browse/KFLUXBUGS-1160[KFLUXBUGS-1160] bug. 
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 6]
```diff
@@ -119,6 +119,52 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
...
+.. Follow Quay's link:https://docs.quay.io/glossary/robot-accounts.html[docs] to create a robot account for your repo.
+.. Give push access to your repository to the robot account.
+.. Retrieve the `.dockercfg` from the "Docker Configuration" menu on your robot account credentials page.
+.. Follow the steps to link:proc_creating-secrets-for-your-builds.adoc[create a secret for your build] and do it as an "Image Pull Secret".
+
+[NOTE] 
+====
+You need to delete the `"auths": {` and one `}` at the end of the docker config file for the secret to work. Until this https://issues.redhat.com/browse/KFLUXBUGS-1160[bug] is resolved. 
+
+For example, this is how you will get the file from Quay:
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:138]:
```suggestion
For example, this is how you get the file from Quay:
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 7]
```diff
@@ -119,6 +119,52 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
...
+    "auth": "example",
+    "email": "example"
+  }
+}
+----
+====
+
+[start=2]
+. Go to the GitHub repo of your component.
+. In each .yaml file in the .tekton directory, under params, locate the param named output-image:
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:165]:
```suggestion
. In each .yaml file within the .tekton directory, under `params`, locate the parameter named `output-image`:
```

File: docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc [hunk 8]
```diff
@@ -119,6 +119,52 @@ To exchange the build task with a memory limit of 6 GB, complete the following s
...
+.. Change it to the new repository. For example: `quay.io/my-new-quay-repo/my-app:on-pr-{\{revision}}` for the `pull-request` pipeline, and `quay.io/my-new-quay-repo/my-app:{\{revision}}` for the `push` pipeline. This is a best practice to know which images come from a Pull Request and which don't.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-to-guides/configuring-builds/proc_customize_build_pipeline.adoc:166]:
```suggestion
.. Change it to relflect your new repository. For example: `quay.io/my-new-quay-repo/my-app:on-pr-{\{revision}}` for the `pull-request` pipeline, and `quay.io/my-new-quay-repo/my-app:{\{revision}}` for the `push` pipeline. This distinction helps identify which images originate from a Pull Request and which do not.

**Review by gtrivedi88 — APPROVED** [review]

**Pull request merged**